packages/opentelemetry-core/src/correlation-context/propagation/HttpCorrelationContext.ts

```diff
@@ -91,13 +91,16 @@ export class HttpCorrelationContext implements HttpTextPropagator {
       return context;
     }
     const pairs = headerValue.split(ITEMS_SEPARATOR);
-    if (pairs.length == 1) return context;
-    pairs.forEach(entry => {
+    for (let i = 0; i < pairs.length; i++) {
```

Comment:
this loops checks only a first pair, so either the loop is not needed, or you should check all pairs and if none of them matched then you should return context

Reply:
That is a good question, as it is now if there is an error in the header it returns the context without correlationContext, as you can see in this test: https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-core/test/correlation-context/HttpCorrelationContext.test.ts#L154 , but may be we would like change this and  continue parsing the other keys and return the well formed keys WDYT?

Reply:
hmm so if I understand correctly only the line `if (pairs.length == 1) return context;` should be removed and the rest stays as it was ?

Reply:
If we want to have a correlation context and only exclude malformed keys , I would say yes, and of change the tests here: https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-core/test/correlation-context/HttpCorrelationContext.test.ts#L154  to reflect that fact.

Reply:
I would like to hear here someone else opinion on that too, if that is the way to go?, @open-telemetry/javascript-approvers

Reply:
I think we should skip malformed pairs and keep processing.